### PR TITLE
Improved focus management test.

### LINKED ISSUE #95: Broken focus management in editor-inline
Probably regression after https://github.com/ckeditor/ckeditor5-link/issues/91.

kapture 2017-04-07 at 14 57 21

### PR DESCRIPTION
**Pull request by oskarwrobel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Improved focus management test. Closes #95.

---

Needs: https://github.com/ckeditor/ckeditor5-ui/pull/194

### CODE REVIEW

**Comment by oleq** [comment]
The editor toolbar remains now, but the focus does not go into the URL input:
kapture 2017-04-12 at 17 36 26

**Comment by oskarwrobel** [comment]
1. Bug is already on master.

2. It's really easy to fix it:
```javascript
this._balloon.add( {
	view: this.formView,
	position: this._getBalloonPositionData()
} );

if ( focusInput ) {
	this.formView.urlInputView.select();
}
```
vs
```javascript
this._balloon.add( {
	view: this.formView,
	position: this._getBalloonPositionData()
} ).then( () => {
	if ( focusInput ) {
		this.formView.urlInputView.select();
	}
} );
```
3. There is a problem with unit tests related to: https://github.com/ckeditor/ckeditor5-ui/issues/203 (test finishes before `promise.then()` is called.

File: tests/link.js [hunk 1]
```diff
@@ -139,7 +139,11 @@ describe( 'Link', () => {
 
 			linkButton.fire( 'execute' );
 
-			expect( selectUrlInputSpy.calledOnce ).to.true;
+			// Focus of url input is called async after internal promise resolve and we are
+			// not able to return this promise.
+			return wait().then( () => {
```
Comment (oskarwrobel) [inline tests/link.js]: It's an ugly hack, I don't like it but I didn't figure out anything less complicated.

**Comment by oskarwrobel** [comment]
I'm refactoring Link tests so let's wait with review till I finish.

**Pull request merged**